packages/@n8n/api-types/src/schemas/insights.schema.ts

```diff
@@ -0,0 +1,44 @@
+import { z } from 'zod';
+
+export const insightsSummaryTypeSchema = z.enum([
+	'total',
+	'failed',
+	'failureRate',
+	'timeSaved',
+	'avgRunTime',
+]);
```

Comment:
could the statuses be uppercased?

and also

```suggestion
const insightsSummaryTypeSchema = z.enum([
	'total',
	'failed',
	'failureRate',
	'timeSaved',
	'avgRunTime',
]);
export const insightsSummaryType = TestCaseExecutionErrorCodes.Values;
```

that way we can use them FE and BE like insightsSummaryType.totla

Reply:
In the db they are just numbers, so from a BE's perspective it does not really matter. Whatever works for you. Maybe we also want to rename `avg` to `average`.

Reply:
@r00gm that part is only FE and will be added in a separate ticket but based on the types so it is still a single source of truth
Here I just wanted a common schema
As @despairblue said they don't need them for the BE

But I can rename `avg` to `average`

Reply:
@r00gm I'll send you the other PR which builds on this and will define the summary types for the FE